internal/telemetry/traces.go

```diff
@@ -33,11 +33,9 @@ func TracerProvider(
 		return noop.NewTracerProvider(), shutdown, nil
 	}
 
-	conn, err := grpc.DialContext(
-		ctx,
+	conn, err := grpc.NewClient(
 		cfg.Traces.Address,
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithBlock(),
 	)
```

Comment:
Same here.

Reply:
It sounds to me like `grpc.WithBlock()` is unnecessary based on https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#especially-bad-using-deprecated-dialoptions.  It sounds like any following RPCs will wait for the connection before proceeding.